### PR TITLE
feat(i18n):i18nify client workbench

### PR DESCRIPTION
**Pull request by rohank07**
### Intent

> Describe briefly what problem this pull request resolves, or what new feature it introduces. Include screenshots of any new or altered UI. Link to any Github issues that are related. 

This PR tackles hard-coded strings to i18n-ify in org.rstudio.core (excluding false-positives).
Issue: #[733 ](https://github.com/StatCan/daaas/issues/733)#[734 ](https://github.com/StatCan/daaas/issues/734)#[735 ](https://github.com/StatCan/daaas/issues/735)#[736](https://github.com/StatCan/daaas/issues/736)

### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

This pull-request and future ones will tackle i18nify classes at package-level (e.g. org.rstudio.core) instead of the per-component approach (like we did for [all-options-menu](https://github.com/rstudio/rstudio/pull/9599))

Implementation follows the same pattern from #9485 as described in gwt/tools/i18n-helpers/README.md. Running the create_dev_locale.sh will create all the new "dev" locale from any new "en" properties files for easier visual confirmation. Refer to README for more information.
### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues. 

Modify an entry in one of the new _en.properties files, and refresh RStudio with ant in devmode.
Run create_dev_locale.sh to clone all _en properties files, which will add a prefix of "@" to each constant and message. Then open RStudio using http://localhost:8787/?locale=dev and browse to the Options->* menus. You can also view in desktop mode by swapping the locale element value attribute from value="en" to value="dev" in RStudio.gwt.xml.

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

# NOTE
I am unsure about the literals in PaneConfig.java, PaneManager.java and RStudioGinModule.java I am referering to the Workbench Tabs (History, Files, Plots etc.). Can you please clarify in which class they can be i18n-able. 

cc @gtritchie

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/workbench/views/terminal/TerminalInfoDialog.java [hunk 1]
```diff
@@ -58,38 +58,38 @@ public TerminalInfoDialog(String globalInfo, final TerminalSession session)
...
-         diagnostics.append("Local-echo:  '").append(localEchoEnabled).append("'\n");
-         diagnostics.append("Working Dir: '").append(cwd).append("'\n");
-         diagnostics.append("Interactive: '").append(cpi.getInteractionModeName()).append("'\n");
-         diagnostics.append("WebSockets:  '").append(userPrefs_.terminalWebsockets().getValue()).append("'\n");
-
-         diagnostics.append("\nSystem Information\n------------------\n");
-         diagnostics.append("Desktop:    '").append(Desktop.isDesktop()).append("'\n");
-         diagnostics.append("Remote:     '").append(Desktop.isRemoteDesktop()).append("'\n");
-         diagnostics.append("Platform:   '").append(BrowseCap.getPlatformName()).append("'\n");
+            diagnostics.append(constants_.busyText()).append(cpi.getHasChildProcs()).append("'\n");
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/terminal/TerminalInfoDialog.java:70]: nitpick: indentation

File: src/gwt/src/org/rstudio/studio/client/workbench/views/presentation2/Presentation2.java [hunk 2]
```diff
@@ -248,4 +249,5 @@ private String asApplicationUrl(String url)
    private final GlobalDisplay globalDisplay_;
    private final EventBus eventBus_;
    private final ApplicationServerOperations server_;
+   private static final PlotsConstants Presentation2Constants = com.google.gwt.core.client.GWT.create(Presentation2Constants.class);
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/presentation2/Presentation2.java:252]: This seems unused

File: src/gwt/src/org/rstudio/studio/client/workbench/views/presentation2/Presentation2.java [hunk 3]
```diff
@@ -30,6 +30,7 @@
 import org.rstudio.studio.client.workbench.views.jobs.events.JobUpdatedEvent;
 import org.rstudio.studio.client.workbench.views.jobs.model.Job;
 import org.rstudio.studio.client.workbench.views.jobs.model.JobConstants;
+import org.rstudio.studio.client.workbench.views.plots.PlotsConstants;
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/presentation2/Presentation2.java:33]: This seems unused

File: src/gwt/src/org/rstudio/studio/client/workbench/views/packages/Packages.java [hunk 4]
```diff
@@ -214,10 +214,8 @@ public void execute(final PackageInstallContext installContext)
             else
             {
                globalDisplay_.showYesNoMessage(MessageDialog.QUESTION,
-                 "Create Package Library",
-                 "Would you like to create a personal library '" +
-                 installContext.getDefaultUserLibraryPath() + "' " +
-                 "to install packages into?",
+                       constants_.createPackageLibraryCaption(),
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/packages/Packages.java:217]: nitpick: indentation

**Review by gtritchie — CHANGES_REQUESTED** [review]
> I am unsure about the literals in PaneConfig.java, PaneManager.java and RStudioGinModule.java I am referering to the Workbench Tabs (History, Files, Plots etc.). Can you please clarify in which class they can be i18n-able.

I need to do some research on this. I suspect this may prove problematic without some additional work and this is very sensitive / fragile code with some non-obvious assumptions.

In meantime here are some smaller tweaks to make.

File: src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java [hunk 5]
```diff
@@ -71,19 +72,19 @@
    public EditSnippetsDialog()
    {
       super(Roles.getDialogRole());
-      setText(constants_.editingEditSnippetsCaption());
```
[the remark this answers is not in the document]
Comment (rohank07) [reply src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java:74]: No, this literal and the others got renamed in SnippetsConstants.java

File: src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java [hunk 6]
```diff
@@ -71,19 +72,19 @@
...
+      setText(constants_.editSnippetsText());
       RStudioGinjector.INSTANCE.injectMembers(this);
       
       addCancelButton();
-      ThemedButton saveButton = new ThemedButton(constants_.editingSaveButtonLabel(), new ClickHandler() {
```
[the remark this answers is not in the document]
Comment (rohank07) [reply src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java:78]: Got renamed and moved to SnippetsConstants.java

File: src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java [hunk 7]
```diff
@@ -71,19 +72,19 @@
...
+      ThemedButton saveButton = new ThemedButton(constants_.saveTitle(), new ClickHandler() {
          public void onClick(ClickEvent event) 
          {
             attemptSaveAndClose();
          }
       });
       addButton(saveButton, ElementIds.DIALOG_OK_BUTTON);
       
-      addLeftWidget(new HelpLink(constants_.editingSnippetHelplink(), "code_snippets"));
```
[the remark this answers is not in the document]
Comment (rohank07) [reply src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java:86]: Got renamed and moved to SnippetsConstants.java

File: src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java [hunk 8]
```diff
@@ -193,7 +194,7 @@ private void attemptSaveAndClose()
             {
                snippetTypes_.setSelectedIndex(i);
                globalDisplay_.showErrorMessage(
-                constants_.errorApplyingSnippetsCaption(snippets.getFileTypeLabel()),
```
[the remark this answers is not in the document]
Comment (rohank07) [reply src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java:196]: Got renamed and moved to SnippetsConstants.java

File: src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (rohank07) [reply src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java:74]: Yes, removed from PrefsConstants.

File: src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (rohank07) [reply src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java:78]: Removed from PrefsConstants

File: src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (rohank07) [reply src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java:86]: Removed from PrefsConstants

File: src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (rohank07) [reply src/gwt/src/org/rstudio/studio/client/workbench/snippets/ui/EditSnippetsDialog.java:196]: Removed from PrefsConstants

**Review by gtritchie — APPROVED** [review]

**Pull request merged**